docs/book/programs/xdp.md

```diff
@@ -1,3 +1,482 @@
 # XDP
 
-This page is a work in progress, please feel free to open a Pull Request!
+!!! example "Source Code"
+
+    Full code for the example in this chapter is available [here](https://github.com/aya-rs/book/tree/main/examples/xdp-drop)
+
+
+## What is XBP in eBPF?
+
+XBP (eXpress Data Path) is a type of eBPF program that attaches to the network interface. It allows for the processing of network packets as soon as they are received from the network driver, even before they enter the networking stack. XDP enables efficient packet filtering, manipulation and redirection at the earliest possible stage, resulting in low-latency and high-throughput.
+
+The idea behind XDP is to add an early hook in the `RX` path of the kernel, and let a user supplied eBPF program decide the fate of the packet. The hook is placed in the NIC driver just after the interrupt processing, and before any memory allocation needed by the network stack itself.
+
+The XDP program is allowed to edit the packet data and, after the XDP program returns, an action code determines what to do with the packet:
+
+* `XDP_PASS`: let the packet continue through the network stack
+* `XDP_DROP`: silently drop the packet
+* `XDP_ABORTED`: drop the packet with trace point exception
+* `XDP_TX`: bounce the packet back to the same NIC it arrived on
+* `XDP_REDIRECT`: redirect the packet to another NIC or user space socket via the [`AF_XDP`](https://www.kernel.org/doc/html/latest/networking/af_xdp.html) address family
+
+XDP requires support in the NIC driver but, as not all drivers support it, it can fallback to a generic operation mode, which performs the eBPF processing in the network stack, though with slower performance.
```

Comment:
I would remove this sentence. You are explaining XDP modes in more detail below, so I think that's sufficient. 🙂

Reply:
Sure, removed the sentence